ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/WithdrawalsProcessor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.core.Withdrawal;
+import org.hyperledger.besu.evm.account.EvmAccount;
+import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+
+import java.util.List;
+
+public interface WithdrawalsProcessor {
+
+  void processWithdrawals(final List<Withdrawal> withdrawals, final WorldUpdater worldUpdater);
+
+  class NoOpWithdrawalsProcessor implements WithdrawalsProcessor {
+
+    @Override
+    public void processWithdrawals(
+        final List<Withdrawal> withdrawals, final WorldUpdater worldUpdater) {}
+  }
+
+  class DefaultWithdrawalsProcessor implements WithdrawalsProcessor {
+
+    @Override
+    public void processWithdrawals(
+        final List<Withdrawal> withdrawals, final WorldUpdater worldUpdater) {
+      for (final Withdrawal withdrawal : withdrawals) {
+        final EvmAccount account = worldUpdater.getOrCreate(withdrawal.getAddress());
+        account.getMutable().incrementBalance(withdrawal.getAmount());
+      }
+      worldUpdater.commit();
```

Comment:
Presumably it's possible for this to throw exceptions? Wonder what happens when it does.

Reply:
It could throw a StorageException, this would be caught by the MainnetBlockValidator and logged. Do you think it would be worth adding some specific log errors for withdrawals?

Reply:
As long as we end up with an invalid block and handle that the right way, not sure if there's value in it being withdrawals specific